feat(core): execution statistics mini chart embed at many place

satvik2131 opened the pull request:
Closes https://github.com/kestra-io/kestra/issues/11126.

## 🧩 Overview

This PR introduces a new reusable component `TimeSeriesChart.vue`, which internally uses `TimeSeries.vue`.

## 🔧 What’s Changed
- Created a new component **`TimeSeriesChart.vue`**, built on top of `TimeSeries.vue`.
- Updated the **hard binding** of `TimeSeries.vue` with the *Flows* page to make it **reusable** across multiple pages.
- Integrated the new component into the following pages:
  - **Dependencies**
  - **Execution Overview**
  - **Triggers**

## 🎥 Demo
https://jam.dev/c/da5fdec1-3cee-4a41-b3b4-001811228e37

## 💬 Next Steps
Looking forward to your suggestions and improvements on this refactor.

satvik2131 committed:
added execution stats settings + updated in the backend request

satvik2131 committed:
fixes

satvik2131 committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

MilosPaunovic committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

MilosPaunovic committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

MilosPaunovic committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

MilosPaunovic committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

MilosPaunovic commented:
Any progress here @satvik2131?

satvik2131 commented:
> Any progress here @satvik2131?

Yes, almost done. Integrating component currently, Will raise a PR today  :)

satvik2131 committed:
TimeSeries Chart component created + integrations

satvik2131 committed:
added TimeSeriesChart at every place mentioned

satvik2131 committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

MilosPaunovic commented:
Just a small note for the future, if you could use **GitHub Keywords** when opening PRs to connect it to the proper issue (to automate the entire process a bit more), it would be super helpful to us. You can read a bit more about it [here](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests#linking-a-pull-request-to-an-issue) (I've added closing keyword for this particular issue).

satvik2131 commented:
> Just a small note for the future, if you could use **GitHub Keywords** when opening PRs to connect it to the proper issue (to automate the entire process a bit more), it would be super helpful to us. You can read a bit more about it [here](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests#linking-a-pull-request-to-an-issue) (I've added closing keyword for this particular issue).

Sure, actually I have re edited the PR readme, so it might got deleted, will take care of that next time, thank you :)
Also you can check the PR now, if it has any issues, please let me know, will fix that.

satvik2131 committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

satvik2131 committed:
added check - if execution happened - show chart in triggers

satvik2131 committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

satvik2131 commented:
@MilosPaunovic, I have added the working demo video, also added a check in triggers, if there is data then only it will show the TimeSeriesChart. Let me know if there are any other fixes, will be working on the other issue - assigned to me :)

satvik2131 committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

MilosPaunovic commented:
@satvik2131 Could you handle the merge conflicts here and after, @Piyush-r-bhaskar will take a look at this PR?

satvik2131 commented:
> @satvik2131 Could you handle the merge conflicts here and after, @Piyush-r-bhaskar will take a look at this PR?

Sure, will do that

satvik2131 committed:
resolved conflicts

satvik2131 committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

satvik2131 commented:
@MilosPaunovic @Piyush-r-bhaskar , please check the PR, I have resolved the conflicts

Piyush-r-bhaskar committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

Piyush-r-bhaskar reviewed (changes requested):
Thank you for this @satvik2131 🚀

But How are you testing this? I don't see any chart now from your branch.

`develop`

<img width="3420" height="1890" alt="Image" src="https://github.com/user-attachments/assets/0799c118-8a09-48f8-9774-f3cb7658ec26" />


`satvik2131:feat(ui)-execution-statistics-component`

<img width="3420" height="1890" alt="Image" src="https://github.com/user-attachments/assets/34fd1435-2c8a-4b71-8393-cbfa11eff5b3" />

satvik2131 commented:
Hi @Piyush-r-bhaskar , I was able to see them after I executed them, like this 
https://jam.dev/c/9cbcc922-4f14-461e-89c9-d1cc47608fce
Please, let me know if you find any issues in it,  I was working with running backend with docker-compose.

Piyush-r-bhaskar commented:
Naah, I still don't anything and there are loads of warning from `TimeSeriesChart.vue` and it's locally on dev port.

<img width="3420" height="1890" alt="image" src="https://github.com/user-attachments/assets/09f4990a-7c43-4a4a-bdb5-a611ccc4b04f" />


@MilosPaunovic  if you get a moment, could you check if it's happening for you or just me ?

satvik2131 committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

satvik2131 commented:
https://github.com/user-attachments/assets/c47d85ca-abad-436f-9783-66a2720a58c8

Here's how I am running it. I am not seeing this much warnings in my machine.

satvik2131 committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

MilosPaunovic commented:
As time allows, I'll be looking at this `PR` next week.

MilosPaunovic committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

MilosPaunovic committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

MilosPaunovic reviewed (changes requested):
Just by opening `Flows` listing I am welcomed with plenty of warnings and errors in the console:

<img width="1914" height="968" alt="image" src="https://github.com/user-attachments/assets/d131c718-24d8-4eee-b337-42777ea2d5d3" />

@satvik2131 This needs to be amended before the review could continue.

MilosPaunovic committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

MilosPaunovic commented:
Hey @satvik2131, are there any updates on this?

satvik2131 commented:
@MilosPaunovic , looking into it.

satvik2131 committed:
solved conflicts

satvik2131 committed:
resolved conflicts

satvik2131 committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

satvik2131 commented:
@MilosPaunovic , can you please check now, previously there was no default execution time set for it. Now, I have fixed it.

satvik2131 committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

MilosPaunovic committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

MilosPaunovic committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

MilosPaunovic committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

MilosPaunovic commented:
Hey @Piyush-r-bhaskar,

Regarding this one, https://github.com/kestra-io/kestra/pull/11738#issuecomment-3461139832, is it solved for you, are you able to do a review of it?

Piyush-r-bhaskar commented:
I will look into it @MilosPaunovic . Meanwhile @satvik2131 can you fix the frontend test failing ?

MilosPaunovic committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

satvik2131 commented:
Sure @Piyush-r-bhaskar , will see why it's failing

MilosPaunovic committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

MilosPaunovic commented:
There seems to be one more merge conflict @satvik2131.

satvik2131 commented:
Hi @MilosPaunovic , will resolve the issues till next week, I am occupied this week.

MilosPaunovic commented:
Not a problem at all.

satvik2131 committed:
resolved conflicts

satvik2131 commented:
Hi @Piyush-r-bhaskar, can you check now, I have updated the TimeSeries import to lazy import, because there was a circular dependency created.

MilosPaunovic committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

Piyush-r-bhaskar committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

MilosPaunovic committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

MilosPaunovic committed:
Merge branch 'develop' into feat(ui)-execution-statistics-component

Piyush-r-bhaskar reviewed (changes requested):
Hey @satvik2131 

- Duration applied from settings not working

<img width="3420" height="1888" alt="Image" src="https://github.com/user-attachments/assets/1406865c-24d0-4b1a-b54b-336d77f43824" />

- can we leave the usage of TimeSeries in Flows.vue as it is , because that is being filtered to show last 30days of executions as applied from config.

-  TimeSeries chart is not working in Sections

<img width="3420" height="1888" alt="Image" src="https://github.com/user-attachments/assets/70fc157f-116f-47ab-b43f-672a2a7a85d1" />

- Days selected from settings is not being applied

MilosPaunovic commented:
Hey @satvik2131, are there any updates on this?

satvik2131 commented:
Hi @MilosPaunovic , will start fixing by tomorrow

satvik2131 commented:
Hi @MilosPaunovic , saw your comment on #11126 , is any new ux already implemented, do i need to fix this PR now ?

MilosPaunovic commented:
Yes, it's implemented, so you can revert all changes made in the Execution Overview page.

Piyush-r-bhaskar commented:
and @satvik2131 , I would love something very similar as done nicely by Milos on Execution Overview chart but only difference is range selector will be in settings.

satvik2131 commented:
Sure @Piyush-r-bhaskar , I am a bit occupied this month, but will surely make time to make these changes ☺️

MilosPaunovic commented:
Hey @satvik2131, 

This PR has diverged too much from the current `develop`, so I'll close it for now and unassign you from the issue.
When you get time to work on it again, feel free to pick it up and raise a fresh PR, unless someone tackles it in the meantime.

Thanks so much for all your help in making Kestra a better product, it's much appreciated! 🚀

satvik2131 commented:
Sure I understand that , no problem will continue afterwards> Hey @satvik2131, 
> 
> This PR has diverged too much from the current `develop`, so I'll close it for now and unassign you from the issue.
> When you get time to work on it again, feel free to pick it up and raise a fresh PR, unless someone tackles it in the meantime.
> 
> Thanks so much for all your help in making Kestra a better product, it's much appreciated! 🚀
